src/typechecker.erl

```diff
@@ -317,24 +317,27 @@ all_type([Ty1|Tys], Ty, AIn, Css, TEnv) ->
     {AOut, Cs} = compat_ty(Ty1, Ty, AIn, TEnv),
     all_type(Tys, Ty, AOut, [Cs|Css], TEnv).
 
-get_record_fields(RecName, Anno, #tenv{records = REnv}) ->
+get_record_fields(RecName, Anno, TEnv) ->
     case typelib:get_module_from_annotation(Anno) of
         {ok, Module} ->
             %% A record type in another module, from an expanded remote type
             case gradualizer_db:get_record_type(Module, RecName) of
                 {ok, TypedRecordFields} ->
                     TypedRecordFields;
                 not_found ->
-                    throw({undef, record, {Module, RecName}})
+                    throw({undef, record, Anno, {Module, RecName}})
             end;
         none ->
             %% Local record type
-            case REnv of
-                #{RecName := Fields} ->
-                    Fields;
-                _NotFound ->
-                    throw({undef, record, RecName})
-            end
+            get_record_fields(RecName, TEnv)
+    end.
+
+get_record_fields(RecName, #tenv{records = REnv}) ->
+    case REnv of
+        #{RecName := Fields} ->
+            Fields;
+        _NotFound ->
+            error({undefined_record, RecName})
```

Comment:
I used `error/1` and not `throw/1` to emphasize that this is invalid input to this function. This is the responsibility of the compiler and gradualizer should only be given forms which don't have undefined records or fields. (I used `undefined_record` and `undefined_field` because these are the atoms the compiler also returns as errors in such cases)

Reply:
Ok, but the compiler *returns* those errors, it doesn't *raise* them as errors, right? There is a difference.... The word "error" is overloaded....

Reply:
yes, yes, there are two things. I looked at the compiler only to choose an atom.
Independent from the compiler, I chose to raise in order to emphasize (that this is not the responsibility of Gradualizer). But I can change this.